Marlin/Marlin_main.cpp

```diff
@@ -600,6 +600,54 @@ void setup_powerhold() {
   #endif
 }
 
+/*
+  Enable 24V to fans, E-reg, Cartridge Holder, Cartridges
+  if no short circuit is detected on line
+  ~200us, blocking time
+  Requires serial to be enabled to generate error condition
+*/
+#if ENABLED(CURRENT_LIMIT)
+void enable_power_supply() {
+	
+  // Turn on FAN first so that 24V does not have a path to 24V_SW through the Rambo. 
+  // TODO: Disconnect these two circuits.
+  pinMode(FAN_CHASSIS_TOP_PIN, OUTPUT);
+  digitalWrite(FAN_CHASSIS_TOP_PIN, HIGH);
+  delay(20);  //ms delay for line to settle. 4*RC = 4*499*10uF = 20ms
+  
+  // Read 24V_SW line voltage with pull-up resistor reference
+  analogRead(PS_MONITOR_PIN);  // Throw out first reading
+  int V_Monitor_Result = analogRead(PS_MONITOR_PIN);
+  
+  if (V_Monitor_Result > PS_ENABLE_LOWER_LIMIT) {     // Test for 24V short to gnd
+    if (V_Monitor_Result <= PS_ENABLE_UPPER_LIMIT) {  // Test for 24V short to 5V
+	  // Test passed - enable power supply
+      digitalWrite(PS_FORCE_ON_LL,LOW); // Pull Power Supply enable low to force on
+      pinMode(PS_FORCE_ON_LL, OUTPUT);  //
```

Comment:
It looks like line 625 and 626 are switched. Shouldn't this be setting `pinMode(PS_FORCE_ON_LL, OUTPUT)` *before* `digitalWrite(PS_FORCE_ON_LL, LOW)` ?